app/upgrades/v28/upgrades.go

```diff
@@ -0,0 +1,70 @@
+package v28
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	stakeibckeeper "github.com/Stride-Labs/stride/v27/x/stakeibc/keeper"
+)
+
+var (
+	UpgradeName = "v28"
+
+	OsmosisChainId = "osmosis-1"
+
+	// Redemption rate bounds updated to give ~3 months of slack on outer bounds
+	RedemptionRateOuterMinAdjustment = sdk.MustNewDecFromStr("0.05")
+	RedemptionRateOuterMaxAdjustment = sdk.MustNewDecFromStr("0.25")
+
+	// Osmosis will have a slighly larger buffer with the redemption rate
+	// since their yield is less predictable
+	OsmosisRedemptionRateBuffer = sdk.MustNewDecFromStr("0.02")
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v27
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	stakeibcKeeper stakeibckeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade v28...")
+
+		// Run migrations first
+		ctx.Logger().Info("Running module migrations...")
+		versionMap, err := mm.RunMigrations(ctx, configurator, vm)
+		if err != nil {
+			return nil, err
+		}
+
+		ctx.Logger().Info("Update redemption rate bounds...")
+		UpdateRedemptionRateBounds(ctx, stakeibcKeeper)
+
+		return versionMap, nil
+	}
+}
+
+// Updates the outer redemption rate bounds
+func UpdateRedemptionRateBounds(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	ctx.Logger().Info("Updating redemption rate outer bounds...")
+
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		// Give osmosis a bit more slack since OSMO stakers collect real yield
+		outerAdjustment := RedemptionRateOuterMaxAdjustment
+		if hostZone.ChainId == OsmosisChainId {
+			outerAdjustment = outerAdjustment.Add(OsmosisRedemptionRateBuffer)
+		}
+
+		outerMinDelta := hostZone.RedemptionRate.Mul(RedemptionRateOuterMinAdjustment)
+		outerMaxDelta := hostZone.RedemptionRate.Mul(outerAdjustment)
+
+		outerMin := hostZone.RedemptionRate.Sub(outerMinDelta)
```

Comment:
Do we need to lower the min? We definitely can but I think we mainly care about the max